alf/utils/math_ops.py

```diff
@@ -198,3 +198,59 @@ def shuffle(values):
     batch_size = alf.nest.get_nest_batch_size(values)
     indices = torch.randperm(batch_size)
     return nest_map(lambda value: value[indices], values)
+
+
+class Softsign_(torch.autograd.Function):
+    r"""Inplace version of softsign function.
+
+    Applies element-wise inplace, the function :math:`\text{SoftSign}(x) = \frac{x}{1 + |x|}`
+    
+    The `current pytorch implementation of softsign
+    <https://pytorch.org/docs/stable/_modules/torch/nn/functional.html#softsign>`_
+    is inefficient for backward because it relies on automatic differentiation
+    and does not have an inplace version. Hence we provide a more efficient
+    implementation.
+
+    Reference:
+    `PyTorch: Defining New Autograd Functions
+    <https://pytorch.org/tutorials/beginner/examples_autograd/two_layer_net_custom_function.html>`_
+    """
+
+    @staticmethod
+    def forward(ctx, input):
+        output = torch.div(input, input.abs() + 1, out=input)
```

Comment:
Just curious, what is ```, out=input)``` used for?

Reply:
the result of the operation will be written to input